framework/src/multiapps/MultiApp.C

```diff
@@ -776,6 +781,64 @@ MultiApp::getCommandLineArgsParamHelper(unsigned int local_app)
     return _cli_args[local_app + _first_local_app];
 }
 
+LocalRankConfig
+rankConfig(unsigned int rank,
+           unsigned int nprocs,
+           unsigned int napps,
+           unsigned int min_app_procs,
+           unsigned int max_app_procs)
+{
+  if (min_app_procs > nprocs)
+    mooseError("minimum number of procs per app is higher than the available number of procs");
+  else if (min_app_procs > max_app_procs)
+    mooseError("minimum number of procs per app must be lower than the max procs per app");
+
+  mooseAssert(rank < nprocs, "rank must be smaller than the number of procs");
```

Comment:
This should probably just be another mooseError

Reply:
I try to leave things that can only be programmer error as assertions.  There should be no way for the user to do anything that triggers this.  If you see the message, a dev wrote bad code. At no point do users enter in any sort of rank information - just number of procs information.  We have loops and such that auto-set the rank to be within some range less than nprocs.

Reply:
This is a stand-alone function that can be called by anyone with any input, I think it should be an error.